MovieQuiz/Presentation/GameRecord.swift

```diff
@@ -0,0 +1,23 @@
+//
+//  GameRecord.swift
+//  MovieQuiz
+//
+//  Created by Виктория Щербакова on 14.11.2022.
+//
+
+import Foundation
+
+/// correct - количество правильных ответов
+/// total - количество вопросов квиза
+/// date - дата завершения раунда
+struct GameRecord: Codable {
+    let correct: Int
+    let total: Int
+    let date: Date
+}
+
+extension GameRecord: Comparable {
+    static func < (lhs: GameRecord, rhs: GameRecord) -> Bool { // рекорд / текущая
+        lhs.correct != rhs.correct ? lhs.correct < rhs.correct : true
```

Comment:
если lhs.correct != rhs.correct - false, то выражение вернет true, что в постановкре lhs < rhs неверно. 
Здесь ошибка, корректно будет оставить только сравнение 
```suggestion
        lhs.correct < rhs.correct
```

Reply:
Большое спасибо за замечания!
Я хотела, чтобы при равных correct сохраненной и текущей игры сохранялся рекорд (и дата) текущей. Что-то похоже есть в примере из документации: https://developer.apple.com/documentation/swift/comparable
```
extension Date: Comparable {
    static func < (lhs: Date, rhs: Date) -> Bool {
        if lhs.year != rhs.year {
            return lhs.year < rhs.year
        } else if lhs.month != rhs.month {
            return lhs.month < rhs.month
        } else {
            return lhs.day < rhs.day
        }
    }
```

Получается, будет лучше вынести проверку на равенство в метод store() ?

Reply:
Ну если исходить из такой задумки, то твоя реализация верная. 

Однако код получился не совсем очевидный, как понимаешь, и об него спотыкаешься. К неочевидному коду как раз принято писать комментарий, почему именно так :) 

Лайк хороший вопрос и предложение. 

Я думаю в таком случае лучше ничего не менять.